.github/workflows/pytest.yml

```diff
@@ -0,0 +1,65 @@
+name: tests
+on:
+  push:
+    branches-ignore:
+
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+  # Runs the workflow once per day at 3:15am
+  schedule:
+    - cron: '3 16 * * *'
+
+env:
+  CACHE_NUMBER: 1 # increase to reset cache manually
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+
+    #outputs:
+    #  branch_coverage: ${{ steps.pytest.outputs.coverage }}
+    #  coverage_report: ${{ steps.pytest.outputs.report }}
+
+    steps:
+      - name: checkout repo content
+        uses: actions/checkout@v2
+
+      - name: Setup Mambaforge
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+            miniforge-variant: Mambaforge
+            miniforge-version: latest
+            activate-environment: fenics-constitutive
+            use-mamba: true
+
+      - name: Set cache date
+        run: echo "DATE=$(date +'%Y%m%d')" >> $GITHUB_ENV
+
+      #- uses: actions/cache@v2
+      #  with:
+      #    path: "/usr/share/miniconda3/envs/fenics-constitutive"
+      #    key: conda-${{ hashFiles('environment.yml') }}-${{ env.DATE }}-${{ env.CACHE_NUMBER }}
+      #  id: cache
+
+      - name: Update environment
+        run: mamba env update -n fenics-constitutive -f environment.yml
```

Comment:
I would say that the `.lock` file is there to _remember_ what the last working configuration was. So It should probably be created after a successful install from the environment file. I am just not sure how often we would want to update it. Should this only be created during after a PR or even after the daily workflow?

Reply:
Using `environment.yml` in the CI would potentially lead to different environments being build due to some deps (of deps) that are not locked. So the daily tests could fail just because the environment could not be solved correctly.

IMO if we want to test the package frequently, then it would make sense to reproduce an environment that is known to be working, i.e. using the lockfile.

The lockfile should be regenerated everytime `environment.yml` has changed. Not sure how to do that with CI, i.e. using the hash of the file or something for [conditional execution](https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution). We could also just do this manually locally with `doit`.

Reply:
But if the environment.yml in the CI is not working , it is also not working for someone that is installing the package from scratch, which is what we want to avoid. One option might be to store the actual configuration (with detailed version numbers of all packages) via github artefacts.

Reply:
In order to check that there is a difference between the completely fixed version setup and the free one (where only the main packages are defined), there could actually be two workflows. From time to time it would then be necessary to manually updated the detailed environment from the running other free version.

Reply:
> But if the environment.yml in the CI is not working , it is also not working for someone that is installing the package from scratch, which is what we want to avoid.

I would not recommend using `environment.yml` to install the package for exactly that reason.

> In order to check that there is a difference between the completely fixed version setup and the free one (where only the main packages are defined), there could actually be two workflows. From time to time it would then be necessary to manually updated the detailed environment from the running other free version.

That's the same what I was trying to say by regenerating the lockfile. This does not have to be done on a daily basis IMO.

Reply:
> One option might be to store the actual configuration (with detailed version numbers of all packages) via github artefacts.

That's exactly what the lockfile is for.

Reply:
I would suggest we move the discussion to a new issue https://github.com/BAMresearch/fenics-constitutive/issues/41. I just had a discussion with Philipp and we are both not really sure where exactly in the actions the lock.file should be used, so this might take a little longer. I would therefore close this discussion and merge because all other points were resolved